include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -18,6 +18,70 @@ def APSIntAttr : Attr<CPred<"$_self.isa<::mlir::IntegerAttr>()">,
   let convertFromStorage = "APSInt($_self.getValue(), !getType().isSigned())";
 }
 
+def SameOperandsIntTypeKind : NativeOpTrait<"SameOperandsIntTypeKind"> {
+  let cppNamespace = "::circt::firrtl";
+}
+
+// A common base class for operations that implement type inference and parsed
+// argument validation.
+class FIRRTLExprOp<string mnemonic, list<OpTrait> traits = []> :
+    FIRRTLOp<mnemonic, traits # [InferTypeOpInterface]> {
```

Comment:
Nice, you can push the `NoSideEffect` trait into this as well

Reply:
Done! :+1: